xla/service/gpu/loop_double_buffer_transformer_test.cc

```diff
@@ -420,8 +418,105 @@ ENTRY main {
   // We expect that the nested while loop has been duplicated, along with its
   // associated computations.
   EXPECT_EQ(while_loops_callees.size(), 6);
+
+  HloInstruction* entry_while_instruction =
```

Comment:
It's unclear to me what coverage the test lines below line 420 add. There are already tests covering this; if the purpose of the test is to check that flattening is preserved, I think it's fine to check only this property (similar for the below test). WDYT?

Reply:
I think that makes sense, i was trying to be strict to check that the unrolling was happening but since the pass wouldnt unroll nested loops, just checking the total number should be fine.